packages/neon_framework/lib/src/models/account.dart

```diff
@@ -26,78 +33,66 @@ abstract interface class Credentials {
 }
 
 /// Account data.
-@JsonSerializable()
 @immutable
-class Account implements Credentials, Findable {
-  /// Creates a new account.
-  Account({
-    required this.serverURL,
-    required this.username,
-    this.password,
-    this.userAgent,
-    @visibleForTesting Client? httpClient,
-  })  : humanReadableID = _buildHumanReadableID(username, serverURL),
-        id = sha1.convert(utf8.encode('$username@$serverURL')).toString() {
-    final cookieStore = NeonStorage().cookieStore(
-      accountID: id,
-      serverURL: serverURL,
-    );
+abstract class Account implements Credentials, Findable, Built<Account, AccountBuilder> {
+  // ignore: public_member_api_docs
+  factory Account([void Function(AccountBuilder)? updates]) = _$Account;
 
-    client = NextcloudClient(
-      serverURL,
-      loginName: username,
-      password: password,
-      appPassword: password,
-      userAgent: userAgent,
-      cookieJar: cookieStore != null ? CookieJarAdapter(cookieStore) : null,
-      httpClient: httpClient,
-    );
-  }
+  const Account._();
 
   /// Creates a new account object from the given [json] data.
-  factory Account.fromJson(Map<String, dynamic> json) => _$AccountFromJson(json);
+  factory Account.fromJson(Map<String, dynamic> json) => _serializers.deserializeWith(serializer, json)!;
 
   /// Parses this object into a json like map.
-  Map<String, dynamic> toJson() => _$AccountToJson(this);
+  Map<String, dynamic> toJson() => _serializers.serializeWith(serializer, this)! as Map<String, dynamic>;
+
+  // ignore: public_member_api_docs
+  static Serializer<Account> get serializer => _$accountSerializer;
 
   @override
-  final Uri serverURL;
+  Uri get serverURL;
   @override
-  final String username;
+  String get username;
   @override
-  final String? password;
+  String? get password;
 
   /// The user agent to use.
-  final String? userAgent;
+  String? get userAgent;
 
-  @override
-  bool operator ==(Object other) =>
-      other is Account &&
-      other.serverURL == serverURL &&
-      other.username == username &&
-      other.password == password &&
-      other.userAgent == userAgent;
-
-  @override
-  int get hashCode => serverURL.hashCode + username.hashCode;
+  /// Custom HTTP client used for all requests.
+  @visibleForTesting
+  @BuiltValueField(serialize: false)
+  Client? get httpClient;
 
   /// An authenticated API client.
-  @JsonKey(includeFromJson: false, includeToJson: false)
-  late final NextcloudClient client;
+  @BuiltValueField(serialize: false)
+  NextcloudClient get client {
```

Comment:
I'm not 100% sure about this one but it was cached before so maybe we should do that again.
So another `@memoized`